src/main/java/com/vanhalen/endpoints/SkittleController.java

```diff
@@ -1,13 +1,38 @@
 package com.vanhalen.endpoints;
 
-import com.vanhalen.logic.SkittleLogic;
+import com.vanhalen.interfaces.SkittleLogicInterface;
+import com.vanhalen.interfaces.MqttServiceInterface;
+import io.swagger.annotations.ApiOperation;
+import org.eclipse.paho.client.mqttv3.MqttException;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
-@RequestMapping("skittles/")
+@RequestMapping("api/skittles")
 public class SkittleController {
+
+    private SkittleLogicInterface _skittleLogic;
+    private MqttServiceInterface _mqttService;
+
     @Autowired
-    private SkittleLogic logic;
+    public SkittleController(SkittleLogicInterface skittleLogic, MqttServiceInterface mqttService){
+        _skittleLogic = skittleLogic;
+        _mqttService = mqttService;
+    }
+
+    @ApiOperation(value = "Use username and password to obtain JWT bearer token", response = String.class)
+    @PostMapping("/sort")
+    public ResponseEntity SortSkittles(@RequestBody String color){
+        try{
+            _mqttService.publishMessage("/VanHalen/color", String.valueOf(color).getBytes());
+            return ResponseEntity.status(HttpStatus.OK).body("Mqtt message sent");
+        } catch (MqttException ex) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ex.getMessage());
+        } catch (Exception ex) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ex.getMessage());
```

Comment:
Is dit voor de end-user niet duplicate code?
Suggestion:
- Gebruik 1 expression

als dit niet kan:
- Zorg dat beide exceptions een andere code versturen